Add new check for discarded futures

markus1189 opened the pull request:
Adds another wart `DiscardedFutures` which is tremendously helpful for us to find places where we introduced `Future`s after changing a blocking method during refactoring.

I don't like that it is almost an exact copy of `NonUnitStatements` but I couldn't find a meaningful way to abstract over it.  I would be happy to try out any suggestions to make this more DRY.

michalrus commented:
@markus1189, aren’t things like

```scala
def foo: Option[Int] = {
  Future(1)
  None
}
```

already caught by `NonUnitStatements`?

markus1189 commented:
Yes but we did not want to have `NonUnitStatements` active because sometimes the external API is such that they become too annoying.  E.g. some mutable collections that return a `boolean` after calling `add`.

ChrisNeveu commented:
This might make more sense as a custom wart for your project since this functionality is already covered by `NonUnitStatements`.

markus1189 commented:
Okay I tried that, but I could not get it working.  I wanted to have the new file inside our project, so I added it in the `project` folder and also added the required dependency to `wartremover`. However wartremover can't find it.  How exaclty would I add this via `wartremoverClasspaths`?

ChrisNeveu commented:
@markus1189 At GMG we have a project for all of our custom warts which we publish as an artifact. You can then add that artifact to your classpath.

markus1189 commented:
@ChrisNeveu I guess that would be a solution, but if possible I would prefer to keep it inside this main project, because the overhead to deploy this additional artifact is quite high

markus1189 commented:
Okay I'll close this then and try to find another way
